language/lang.go

```diff
@@ -99,13 +99,21 @@ type Language interface {
 // FinishableLanguage allows a Language to be notified when Generate is finished
 // being called.
 type FinishableLanguage interface {
+	// Init is called before any calls to GenerateRules
+	// This allows for hooks to be called, for instance to starting a background
+	// server process.
+	Init()
```

Comment:
Let's pass a context here so that requests can be killed when Gazelle shuts down